✨ FIRST_TIME composition api (#5029)
kodadot/nft-gallery: FIRST_TIME.md

**1. How can I read some data from the GraphQL?**

Every `.graphql` file is located in the `queries/`.

```graphql
query nftByIdMinimal($id: String!) {

>   nft: nFTEntity(id: $id) {

Reviewer: ```suggestion
  nft: nftEntityById(id: $id) {
```

    id
    currentOwner
    price
    events
  }
}
```

Then we can use it like this:

```html
<script lang="ts" setup>
const route = useRoute()
const { data: nft } = useGraphql({

>   queryName: 'nftByIdMinimal',

Author: ```suggestion
  queryName: 'nftById',
```

  variables: { id: route.params.id },
})

console.log(nft)
</script>
```

**2. How can I read on-chain data from the RPC node?**

```html
<script lang="ts" setup>
const { apiInstance } = useApi()

const collectionId = ref('0')
const id = ref('0')
const api = await apiInstance.value
const nft = await api.query.uniques.asset(collectionId, id)

> console.log(nft)

Reviewer: use consola

</script>
```

**3. How can I make an on-chain transaction?**

```html
<script lang="ts" setup>
async function submit() {
  const { accountId } = useAuth()
  const { howAboutToExecute, isLoading, status, initTransactionLoader } =
    useMetaTransaction()

  const executeTransaction = ({ cb, arg, successMessage, errorMessage }) => {
    initTransactionLoader()
    howAboutToExecute(
      accountId.value,
      cb,
      arg,
      () => infoMessage(successMessage || 'Success!'),
      () => dangerMessage(errorMessage || 'Failed!')
    )
  }

  executeTransaction({
    cb: api.tx.system.remark,
    arg: ['args']
  })
}
</script>
```

**4. How can I test Kodadot without spending KSM?**

For Kusama:

> [You can obtain some Westend (WND)](https://matrix.to/#/#westend_faucet:matrix.org)

Reviewer: remove

For Basilisk (Rococo):
[You can obtain some KSM & BSX](https://discord.com/channels/840514076538830888/881839877140930630)

You can change the network in the navbar.

> Currently supported networks are `Basilisk, Kusama, Westend, statemine, westmint`.

Reviewer: ```suggestion
Currently supported networks are `Basilisk`, `Basilisk-Rococo` and `Kusama`.
EVM chains such as `MoonBeam` and `MoonRiver` are in read-only mode.

```

Current Indexers we have/use:

- SubSquid
  - Basilisk: [snek](https://github.com/kodadot/snek)
  - Kusama: [rubick](https://github.com/kodadot/rubick)

>   - Moonsama: [click](https://github.com/kodadot/click)

Reviewer: ```suggestion
  - MoonRiver: [click](https://github.com/kodadot/click)
```
